src/VisualStudio/Core/Impl/SolutionExplorer/SymbolTree/RootSymbolTreeItemSourceProvider.cs

```diff
@@ -211,5 +213,29 @@ void OnItemPropertyChanged(object sender, PropertyChangedEventArgs e)
                 }
             }
         }
+
+        void AddToDictionary(string currentFilePath, RootSymbolTreeItemCollectionSource source)
+        {
+            if (!_filePathToCollectionSources.TryGetValue(currentFilePath, out var sources))
+            {
+                sources = [];
+                _filePathToCollectionSources[currentFilePath] = sources;
+            }
+
+            sources.Add(source);
+        }
+
+        void RemoveFromDictionary(string currentFilePath, RootSymbolTreeItemCollectionSource source)
```

Comment:
(feel free to ignore since this is impacting RPS)

...and MultiRemove from the same file as above.